examples/determined-ai/README.md

```diff
@@ -0,0 +1,326 @@
+# Determined AI Example
+
+[Determined AI](https://docs.determined.ai/latest/index.html)
+
+This example shows how to package an application as a KOTS app using the [Replicated Library Chart](https://github.com/replicatedhq/helm-charts/tree/main/charts/replicated-library). Included are several practical examples of common patterns used when deploying a helm chart with KOTS.
+
+# Table of Contents
+* [Examples](#examples)
+    * [Embedded vs. External Database](#embedded-vs-external-database)
+    * [Self-signed Ingress TLS Certificate vs. User-provided](#self-signed-ingress-tls-certificate-vs-user-provided)
+    * [Pass Labels and Annotations from Config Options to Helm Chart Values](#pass-labels-and-annotations-from-config-options-to-helm-chart-values)
+    * [Wait for Database to Start Before Starting your Application](#wait-for-database-to-start-before-starting-your-application)
+
+# Examples
+
+All of the below examples are snippets from the full example chart in this directory
+
+## Embedded vs. External Database
+
+You may have the need to allow end-users to choose between hosting the database required by your application themselves or providing one with the application at install time. This example will show you how to package templates in your helm chart to allow this functionality for a Postgres database as well as how you would integrate it with your KOTS app.
+
+1. First you'll need to template in your helm chart the optionality between "embedded" vs. "external" Postgres
+
+[values.yaml](values.yaml)
+```yaml
+postgresql:
+  # This is a subchart that we're importing from Bitnami
+  enabled: true
+  image:
+    registry: docker.io
+    repository: bitnami/postgresql
+    tag: 15.3.0-debian-11-r0
+  fullnameOverride: postgresql
+  auth:
+    postgresPassword: determined
+
+determined:
+  # Here is where we defined our custom values to implement our templating
+  postgresPassword: determined
+  externalPostgres:
+    enabled: false
+    username: postgres
+    password: determined
+    database: postgres
+    host: postgresql
+    port: 5432
+```
+
+**NOTE**: Determined AI uses a secret containing a config file where all of the application configuration lives including the database connection string. The below is a snippet from that configuration file where we are adding our templating. For you this same templating might be added in environment variables or elsewhere but the same logic applies.
+
+[replicated-library.yaml](templates/replicated-library.yaml)
+```yaml
+secrets:
+  determined:
+    data:
+      master.yaml: |
+      ...
+        db:
```

Comment:
This isn't valid yaml, I'm not sure what the intention was. Should the pipe be here?
Anytime we have a code snippet I think it's important that it be fully functioning.

Reply:
Yeah, i'm using the `...` in areas where i'm not providing the full copy of the example and attempting to highlight the specific relevant bit from the larger file. I could transition to making sure that each snippet of code could technically be functional, but it may end up needing to include more than what the specific example is trying to highlight. Above each snippet I do link to the file where the snippet comes from that does include the full working example.  

Open to suggestions here

Reply:
Open to getting rid of those placeholders, but just curious to know if you're expecting each code snippet to be copy/pasteable into a new file.

Reply:
Looking through the example I see what you've done here. I think in the code block I would just remove the `...` so the example is valid yaml. You can link to the full example if someone wants to see the whole thing. I'm just thinking of someone that doesn't know yaml and ends up copying that `...` and looses a couple of days trying to figure out why it isn't working 🤣